src/main/java/com/mraof/minestuck/item/BoondollarsItem.java

```diff
@@ -29,9 +29,11 @@ public BoondollarsItem(Properties properties)
 	@Override
 	public InteractionResultHolder<ItemStack> use(Level level, Player playerIn, InteractionHand handIn)
 	{
-		if(!level.isClientSide && !(playerIn instanceof FakePlayer))
+		if(playerIn instanceof ServerPlayer serverPlayer)
```

Comment:
No. A `FakePlayer` is definitely an instance of `ServerPlayer` still.
Rather, it has become redundant in some places by actually checking the return value of  `PlayerSavedData.getData(ServerPlayer)`/`PlayerData.get(ServerPlayer)`. I changed the nullable return of that function to an optional, and along with it also changed a lot of places to use `ifPresent()` instead of throwing, like here. Because the data is null/missing specifically for fake players, this works out to make most `FakePlayer` checks redundant.